app/forms/developer_search_form.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class DeveloperSearchForm
+  include Virtus.model
+  include ActiveModel::Model
+
+  attribute :programming_language_id, Integer
+  attribute :language_id, Integer
+
+  def search
```

Comment:
I wonder if you don't see any inefficient queries in the log.
Can you add 100 developers records, for instance?
Then do the search and check the log, `log/development.log`?

Reply:
Added eager loading to fix this!